app/plugin.go

```diff
@@ -165,29 +166,36 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 
 	prepackagedPluginsDir, found := fileutils.FindDir("prepackaged_plugins")
 	if found {
-		if err := filepath.Walk(prepackagedPluginsDir, func(walkPath string, info os.FileInfo, err error) error {
-			if !strings.HasSuffix(walkPath, ".tar.gz") {
-				return nil
+		var walkPaths []string
+		filepath.Walk(prepackagedPluginsDir, func(walkPath string, info os.FileInfo, err error) error {
+			if strings.HasSuffix(walkPath, ".tar.gz") {
+				walkPaths = append(walkPaths, walkPath)
 			}
+			return nil
+		})
 
-			fileReader, err := os.Open(walkPath)
-			if err != nil {
-				mlog.Error("Failed to open prepackaged plugin", mlog.Err(err), mlog.String("path", walkPath))
-				return nil
-			}
-			defer fileReader.Close()
+		var wg sync.WaitGroup
```

Comment:
Will this approach to synchronization be a problem if there were dozens of prepackagaed plugins, and all get installed at the same time?

Reply:
I tried testing with 5 plugins and it worked fine.